app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt

```diff
@@ -1149,5 +1221,102 @@ class HomeFragment : Fragment() {
 
         // Elevation for undo toasts
         internal const val TOAST_ELEVATION = 80f
+
+        private const val POCKET_STORIES_TO_SHOW_COUNT = 7
+        private const val POCKET_CATEGORIES_SELECTED_AT_A_TIME_COUNT = 7
+    }
+}
+
+/**
+ * Get the list of stories to be displayed.
+ * Either the stories from the [POCKET_STORIES_DEFAULT_CATEGORY_NAME] either
+ * filtered stories based on the user selected categories.
+ *
+ * @param allCategories the list of all stories Pocket stories categories.
+ * @param selectedCategories the list of the currently user selected categories.
+ * @param neededStoriesCount how many stories are intended to be displayed.
+ * This only impacts filtered results guaranteeing an even spread of stories from each category.
+ */
+@VisibleForTesting
+internal fun getFilteredStories(
```

Comment:
It looks like those helpers here are independent from the `HomeFragment` class. Maybe we can move them somewhere else. There's a high risk that the `HomeFragment` will grow a lot. :)

Reply:
Maybe into a `Controller` 😅? 
Was thinking about an intermediate big interface for all domain logic needed for home composables, something like `HomeComposablesLogic`?(please suggest a better name) with a default implementation that can be delegated to by the HomeFragment? 
This just while migrating current views to composables, though even after fully migrating this screen to Compose, if bringing back all this logic in one place will have that again being immense?

Reply:
I don't have a strong opinion on the naming. It just seems like some utility functions (or maybe they could even be extension functions?) that could move to some related utilities file. Just to keep the fragment focused.